pkg/webhook/nad/mutator.go

```diff
@@ -0,0 +1,66 @@
+package nad
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	cniv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	"github.com/yaocw2020/webhook/pkg/types"
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/harvester/harvester-network-controller/pkg/network/iface"
+	"github.com/harvester/harvester-network-controller/pkg/utils"
+)
+
+type nadMutator struct {
+	types.DefaultMutator
+}
+
+var _ types.Mutator = &nadMutator{}
+
+func NewNadMutator() *nadMutator {
+	return &nadMutator{}
+}
+
+func (n *nadMutator) Create(request *types.Request, newObj runtime.Object) (types.Patch, error) {
+	netAttachDef := newObj.(*cniv1.NetworkAttachmentDefinition)
```

Comment:
check type converting, or assume it is always `cniv1.NetworkAttachmentDefinition` ?

Reply:
Yes, it should be always cniv1.NetworkAttachmentDefinition. If not and something happen out of expectation, it's better to fix it in the dependency package.